zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/ZkClient.java

```diff
@@ -984,11 +1000,43 @@ private void fireAllEvents() {
 
   protected List<String> getChildren(final String path, final boolean watch) {
     long startT = System.currentTimeMillis();
+    // Need one element array to change value of this final variable.
+    final int[] connectionLossRetryCount = {0};
```

Comment:
? this pattern looks very confusing,

Reply:
This variable needs to be accessed in inner class Callable so it needs to be final.
And since it is final, we could not increment the value if we make a an integer. So we need this one element array to change value of this final variable. I understand it looks a bit ugly, but since we are removing the code part in a few months once we have a better way in zk server, I wanted to make it simple without creating a wrapper class.

Reply:
Can we use wrapped integer?

Reply:
Also, connectionLossRetryCount seems only used within callable, why not define within the callable function?

Reply:
Considering clean code, I think we could implement a GetChildrenCallable class that has the counter internally, so we could get rid of the one element array. It will look better

Reply:
Resolved. I was thinking in the lambda way that we could not define the private member in callable class but actually we could when we new a callable class.. Kind of missing the old stuff.